etcd/v3/kv_etcd.go

```diff
@@ -1427,20 +1427,29 @@ func (et *etcdKV) addMember(
 	}
 	resp := make(map[string][]string)
 	ctx, cancel = et.MaintenanceContextWithLeader()
-	memberListResponse, err := et.kvClient.MemberList(ctx)
-	cancel()
-	if err != nil {
-		return nil, err
-	}
-	for _, member := range memberListResponse.Members {
-		if member.Name == "" {
-			// Newly added member
-			resp[nodeName] = member.PeerURLs
-		} else {
-			resp[member.Name] = member.PeerURLs
+	addMemberListRetryCount := 3
+	// Retry memberlist because of https://github.com/etcd-io/etcd/issues/11198
+	// Above issue fixed in etcd v3.5 whereas our current embeded etcd is v3.4.24.
+	for i := 0; i < addMemberListRetryCount; i++ {
+		memberListResponse, err := et.kvClient.MemberList(ctx)
+		cancel()
+		if err != nil {
+			return nil, err
+		}
+		for _, member := range memberListResponse.Members {
+			if member.Name == "" {
+				// Newly added member
+				resp[nodeName] = member.PeerURLs
+			} else {
+				resp[member.Name] = member.PeerURLs
+			}
 		}
+		if _, ok := resp[nodeName]; ok {
+			return resp, nil
+		}
+		time.Sleep(10 * time.Second)
```

Comment:
This sleep seems too long. Also, I feel that we should not retry. Instead, just log a warning and add the entry into our map ourselves.

Reply:
Done